tests/Common/WebFrameworkTestCase.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace DDTrace\Tests\Common;
+
+use DDTrace\Tests\Frameworks\Util\CommonScenariosDataProviderTrait;
+use DDTrace\Tests\Frameworks\Util\Request\GetSpec;
+use DDTrace\Tests\Frameworks\Util\Request\RequestSpec;
+use DDTrace\Tests\WebServer;
+
+/**
+ * A basic class to be extended when testing web frameworks integrations.
+ */
+abstract class WebFrameworkTestCase extends IntegrationTestCase
+{
+    use CommonScenariosDataProviderTrait;
+
+    const PORT = 9999;
+
+    /**
+     * @var WebServer|null
+     */
+    private static $appServer;
+
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        static::setUpWebServer();
+    }
+
+    public static function tearDownAfterClass()
+    {
+        parent::tearDownAfterClass();
+        static::tearDownWebServer();
+    }
+
+    /**
+     * Returns the application index.php file full path.
+     *
+     * @return string|null
+     */
+    protected static function getAppIndexScript()
+    {
+        return null;
+    }
+
+    /**
+     * Get additional envs to be set in the web server.
+     * @return array
+     */
+    protected static function getEnvs()
+    {
+        return [];
+    }
+
+    /**
+     * Sets up a web server.
+     */
+    protected static function setUpWebServer()
+    {
+        $rootPath = static::getAppIndexScript();
+        if ($rootPath) {
+            self::$appServer = new WebServer($rootPath, '0.0.0.0', self::PORT);
+            self::$appServer->setEnvs(static::getEnvs());
+            self::$appServer->start();
+        }
+    }
+
+    /**
+     * Tear down the  web server.
+     */
+    private static function tearDownWebServer()
+    {
+        if (self::$appServer) {
+            self::$appServer->stop();
+        }
+    }
+
+    /**
+     * Executed a call to the test web server.
+     *
+     * @param RequestSpec $spec
+     * @return mixed|null
+     */
+    protected function call(RequestSpec $spec)
+    {
+        $url = 'http://127.0.0.1:' . self::PORT . $spec->getPath();
+        if ($spec instanceof GetSpec) {
+            $response = $this->sendRequest('GET', $url);
+            return $response;
+        } else {
```

Comment:
This was intended because at the moment we only support GetSpec, so is someone implement a PutSpec and send down the pipe we give notice that he/she has  something else to add.
Does it make sense? :)

Reply:
Yes, but I'm saying we don't need to wrap it with `else { /* ... */ }` since the supported `GetSpec` has already returned at this point. So we can just fall into `$this->fail()`. :)

Reply:
Ahhhh you are right :D